xbean-finder/src/test/java/org/apache/xbean/finder/archive/JarArchiveTest.java

```diff
@@ -95,14 +96,65 @@ public void testIterator() throws Exception {
             actual.add(entry.getName());
         }
 
-        assertFalse(0 == actual.size());
+        Assert.assertFalse(0 == actual.size());
 
         for (Class clazz : classes) {
-            assertTrue(clazz.getName(), actual.contains(clazz.getName()));
+            Assert.assertTrue(clazz.getName(), actual.contains(clazz.getName()));
         }
 
-        assertEquals(classes.length, actual.size());
+        Assert.assertEquals(JarArchiveTest.classes.length, actual.size());
     }
 
+    @Test
+    public void testXBEAN337() throws Exception {
```

Comment:
Maybe get the missing cases from the diff, it covers a bit more the pr cases IIRC

Reply:
well, i don't completely understand where i should look to... looking into pulled tests in branch and see nothing.
could you please point?

Reply:
this test (https://gist.github.com/rmannibucau/ced8a809672a12a6f026e5ccac58a679#file-xbean-exclamation-mark-diff-L200) was covering 5 use cases, I don't think them all in this test

Reply:
thanks, they helped to find/fix 1 issue.
can i ask you to integrate them?
cannot force my git to apply patch, so have to do it manually. and afraid it will turn in too much cosmetics to fix...

Reply:
Sure - but will likely wait tomorrow.
Just to be sure: you want I integrate the test as disabled on main branch?

Reply:
well, what is easier for you - disabled on main branch so i sync, merge locally, enable them and run; or to this branch, as enabled.

Reply:
Just pushed on trunk branch with `@Ignore("PR-34")`.

Reply:
> Just pushed on trunk branch with `@Ignore("PR-34")`.

thanks. thanks a lot!

![tests](https://user-images.githubusercontent.com/2069825/208536744-7ead54c2-3522-4f6f-a5f5-bca0800f1be8.png)